File: src/main/java/net/greemdev/core/commands/NameCommand.java
package net.greemdev.core.commands;

import net.greemdev.core.util.CommandUtil;
import org.bukkit.Bukkit;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class NameCommand implements CommandExecutor {
    @Override
    public boolean onCommand(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (CommandUtil.warnIfNotAuthorized(sender, "core.name")) return true;
        if (CommandUtil.warnIfConsole(sender)) return true;

        Player player = Objects.requireNonNull(CommandUtil.asPlayer(sender));

        ItemStack item = player.getInventory().getItemInMainHand();
        ItemMeta meta = item.getItemMeta();

        if (CommandUtil.warnIf(sender, meta == null, "You're not holding anything.") ||
                CommandUtil.warnIfEmptyArgs(sender, args)) {
            return true;
        }

        Objects.requireNonNull(meta).setDisplayName(String.join(" ", args)); //meta never can be null at this point but IJ was screaming at me so here you go
        item.setItemMeta(meta);
        player.sendMessage(ChatColor.DARK_AQUA + "Successfully set the item's name in your main hand to \"" + String.join(" ", args) + "\".");
        return true;
    }
}

File: src/main/kotlin/net/greemdev/core/commands/NameCommand.kt
package net.greemdev.core.commands

import net.greemdev.core.objects.CommandEvent
import net.greemdev.core.objects.CommandResult
import net.greemdev.core.util.*
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

public class NameCommand: CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        return this.analyzeResult(this.executeCommand(CommandEvent.from(sender, command, label, args)))
    }

    private fun executeCommand(event: CommandEvent): CommandResult {
        if (event.sender.warnIfNotAuthorized("core.name")
            || event.sender.warnIfConsole()) return CommandResult.unsuccessful()

        val item = event.player.inventory.itemInMainHand
        val meta = item.itemMeta

        if (event.sender.warnIf(meta == null, "You're not holding anything.") ||
                event.sender.warnIfEmptyArgs(event.args)) return CommandResult.unsuccessful()

        meta.setDisplayName(event.args.join(" "))
        item.itemMeta = meta
        event.player.sendCoreMessage(ChatColor.DARK_AQUA.toString() + "Successfully set the item's name in your main hand to \"" + event.args.joinToString(separator = " ") + "\".")
        return CommandResult.successful()
    }
}